File: gitdot-web/packages/api/src/endpoint/repository/list-latest-repositories.ts
import { z } from "zod";
import { RepositoryResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListLatestRepositoriesRequest = z.object({});
export type ListLatestRepositoriesRequest = z.infer<
  typeof ListLatestRepositoriesRequest
>;

export const ListLatestRepositoriesResponse = z.array(RepositoryResource);
export type ListLatestRepositoriesResponse = z.infer<
  typeof ListLatestRepositoriesResponse
>;

export const ListLatestRepositories = {
  path: "/repository/latest",
  method: "GET",
  request: ListLatestRepositoriesRequest,
  response: ListLatestRepositoriesResponse,
} as const satisfies Endpoint;
export type ListLatestRepositories = typeof ListLatestRepositories;

File: gitdot-api/src/endpoint/repository/list_latest_repositories.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryResource};

pub struct ListLatestRepositories;

impl Endpoint for ListLatestRepositories {
    const PATH: &'static str = "/repository/latest";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListLatestRepositoriesRequest;
    type Response = ListLatestRepositoriesResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListLatestRepositoriesRequest {}

pub type ListLatestRepositoriesResponse = Vec<RepositoryResource>;
